src/main/java/io/confluent/idesidecar/restapi/resources/RestProxyResource.java

```diff
@@ -43,16 +64,78 @@ public void schemaRegistryClusterProxy(RoutingContext routingContext) {
   }
 
   private void handleClusterProxy(RoutingContext routingContext, ClusterProxyContext proxyContext) {
-    clusterProxyProcessor.process(proxyContext)
+    process(routingContext, clusterProxyProcessor, proxyContext);
+  }
+
+  private void handleRBACProxy(RoutingContext routingContext, ProxyContext proxyContext) {
+    process(routingContext, rbacProxyProcessor, proxyContext);
+  }
+
+  public record RBACRequest(
+      @NotNull String userPrincipal,
+      @NotNull Action[] actions
+  ) {
+    public record Action(
+        @NotNull String resourceType,
+        @NotNull String resourceName,
+        @NotNull String operation,
+        @NotNull Scope scope
+    ) {
+      public record Scope(
+          @NotNull Map<String, String> clusters,
+          @NotNull String[] path
+      ) {}
+    }
+  }
```

Comment:
This looks great! I think we should remove the `@NotNull` constraints since we don't really utilize the request body anywhere but to generate the OpenAPI spec.